actions/checkForAndPlaceOrder.ts

```diff
@@ -128,10 +133,14 @@ const _checkForAndPlaceOrder: ActionFn = async (
       // Check if the order is due (by blockNumber)
       if (
         lastResult?.result === PollResultCode.TRY_ON_BLOCK &&
-        lastResult.blockNumber < blockNumber
+        blockNumber < lastResult.blockNumber
```

Comment:
this too! we really need unit testing and write this in a more functional approach to be able to test it easier